lib/rust/enso-font/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+//! The Enso Font. This crate supports downloading and unpacking the font family, as well as
+//! constructing a reduced font family from a subset of the fonts.
+
+// === Features ===
+#![feature(let_chains)]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+#![allow(clippy::bool_to_int_with_if)]
+#![allow(clippy::let_and_return)]
+// === Non-Standard Linter Configuration ===
+#![deny(unconditional_recursion)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+
+use ide_ci::prelude::*;
+pub use owned_ttf_parser as ttf;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const ENSO_FONT_FAMILY_NAME: &str = "enso";
+
+const ENSO_FONT_FAMILY_FONTS: &[Font] = &[
+    Font { name: "Black", weight: ttf::Weight::Black },
+    Font { name: "Bold", weight: ttf::Weight::Bold },
+    Font { name: "ExtraBold", weight: ttf::Weight::ExtraBold },
+    Font { name: "ExtraLight", weight: ttf::Weight::ExtraLight },
+    Font { name: "Light", weight: ttf::Weight::Light },
+    Font { name: "Medium", weight: ttf::Weight::Medium },
+    Font { name: "Regular", weight: ttf::Weight::Normal },
+    Font { name: "SemiBold", weight: ttf::Weight::SemiBold },
+    Font { name: "Thin", weight: ttf::Weight::Thin },
+];
+
+/// The URL for the Enso Font package, excluding the final component.
+pub const PACKAGE_BASE_URL: &str = "https://github.com/enso-org/font/releases/download/1.0/";
+/// The final (filename) component of the URL for the Enso Font package.
+pub const PACKAGE_FILE: &str = "enso-font-1.0.tar.gz";
+const PACKAGE_FONTS_PREFIX: &str = "ttf";
+
+
+
+// ===================
+// === Font Family ===
+// ===================
+
+/// A collection of fonts.
+#[derive(Debug)]
+pub struct FontFamily {
+    fonts: Vec<Font>,
+}
+
+impl FontFamily {
+    /// Returns the Enso Font.
+    pub fn enso() -> Self {
+        FontFamily { fonts: ENSO_FONT_FAMILY_FONTS.to_vec() }
+    }
```

Comment:
As we in `enso-enso-font` crate, I assume it's only about "Enso" font, so this could be called just `new`.